core/evalir.ml

```diff
@@ -585,6 +585,25 @@ struct
              | `Bool true     -> t
              | `Bool false    -> e
              | _              -> eval_error "Conditional was not a boolean")
+
+  and eval_nested_query env cont (db: Value.database) pkg =
```

Comment:
Yes, it's entirely because of nested records: the normal forms are nested (since the results are of the form `(!data = ..., !from = ..., !to)` (a la the Provenance translation Stefan did). The actual querying can probably be done with the flat policy; I think I insisted on nested-ness for typechecking.

We'll need the full generality of shredding when we think of shredded temporal queries, but I think that's along the line a bit.